### PR TITLE
[WIP] changefeed, processoe (ticdc): make ddl only block related tables

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #xxx

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/model/sink.go [hunk 1]
```diff
@@ -194,6 +195,37 @@ type TableName struct {
 	IsPartition bool   `toml:"is-partition" json:"is-partition" msg:"is-partition"`
 }
 
+func (t *TableName) MarshalJSON() ([]byte, error) {
```
Comment (CharlesCheung96) [inline cdc/model/sink.go:198]: Why not just use `json.Marshal(t)`?

File: cdc/owner/changefeed.go [hunk 2]
```diff
@@ -774,158 +784,87 @@ func (c *changefeed) preflightCheck(captures map[model.CaptureID]*model.CaptureI
...
 		c.barriers.Update(syncPointBarrier, nextSyncPointTs)
 	case finishBarrier:
-		if fullyBlocked {
+		if checkpointReachBarrier {
 			c.feedStateManager.MarkFinished()
 		}
-		return barrierTs, nil
+		break
+	case ddlResolvedBarrier:
+		c.barriers.Update(ddlResolvedBarrier, c.ddlPuller.ResolvedTs())
```
Comment (CharlesCheung96) [inline cdc/owner/changefeed.go:861]: Redo ddlResolvedTs is also a type of ddlResolvedBarrier,  so should we write DDL events to redo in ddl_holder and get ddl-resolvedTs from `ddl_holder`?  
In addition, encapsulating all schema and DDL related logic inside `ddl_holder` would be a better way to improve code maintainability?

File: cdc/model/sink.go (hunk 1 above)
Comment (asddongmen) [reply cdc/model/sink.go:199]: sure

File: cdc/owner/changefeed.go (hunk 2 above)
Comment (asddongmen) [reply cdc/owner/changefeed.go:861]: good idea

**Pull request closed without merging**